### PR TITLE
Controller runtime 0.8.3, CRD v1, Go 1.16

### PR DESCRIPTION
**Pull request by pepov**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | yes
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?
Upgrades controller-runtime and k8s libraries. Upgrades controller-gen and CRDs to v1 from v1beta1 (requires k8s 1.16 or later). Upgrade to go 1.16.

### Why?
To stay up to date and to avoid k8s warnings because of v1beta1 CRDs.

### CODE REVIEW

File: pkg/sdk/api/v1beta1/groupversion_info.go [hunk 1]
```diff
@@ -18,16 +18,18 @@
 package v1beta1
 
 import (
+	"k8s.io/apimachinery/pkg/runtime"
```
Comment (siliconbrain) [inline pkg/sdk/api/v1beta1/groupversion_info.go:21]:
```suggestion
```

File: pkg/sdk/api/v1beta1/groupversion_info.go [hunk 2]
```diff
@@ -18,16 +18,18 @@
...
 	"k8s.io/apimachinery/pkg/runtime/schema"
-	"sigs.k8s.io/controller-runtime/pkg/runtime/scheme"
+	// "sigs.k8s.io/controller-runtime/pkg/runtime/scheme"
```
Comment (siliconbrain) [inline pkg/sdk/api/v1beta1/groupversion_info.go:23]:
```suggestion
	"sigs.k8s.io/controller-runtime/pkg/scheme"
```
I think we should still use `controller-runtime`s scheme builder.

File: pkg/sdk/api/v1beta1/suite_test.go [hunk 3]
```diff
@@ -15,24 +15,19 @@
...
-	"go.uber.org/zap/zapcore"
 	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
 
 	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/envtest"
-
-	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	// logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
```
Comment (siliconbrain) [inline pkg/sdk/api/v1beta1/suite_test.go:30]:
```suggestion
```

File: pkg/sdk/api/v1beta1/suite_test.go [hunk 4]
```diff
@@ -53,12 +48,13 @@ func TestAPIs(t *testing.T) {
...
-		panic(fmt.Sprintf("who watches the watchmen (%v)?", err))
-	}
-	log := zapr.NewLogger(zapLog)
-	logf.SetLogger(log)
+	// zapLog, err := zap.NewDevelopment(zap.ErrorOutput(zapcore.AddSync(GinkgoWriter)))
+	// if err != nil {
+	// 	panic(fmt.Sprintf("who watches the watchmen (%v)?", err))
+	// }
+	// log := zapr.NewLogger(zapLog)
+	// logf.SetLogger(log)
```
Comment (siliconbrain) [inline pkg/sdk/api/v1beta1/suite_test.go:56]:
```suggestion
```

File: controllers/logging_controller.go [hunk 5]
```diff
@@ -175,7 +176,7 @@ func (f *secretLoaderFactory) OutputSecretLoaderForNamespace(namespace string) s
 // SetupLoggingWithManager setup logging manager
 func SetupLoggingWithManager(mgr ctrl.Manager, logger logr.Logger) *ctrl.Builder {
 	requestMapper := handler.EnqueueRequestsFromMapFunc(func(obj client.Object) []reconcile.Request {
-		object, err := meta.Accessor(obj.DeepCopyObject())
+		object, err := meta.Accessor(runtime.Object(obj))
```
Comment (siliconbrain) [inline controllers/logging_controller.go:179]:
```suggestion
		object, err := meta.Accessor(obj)
```

**Review by bshifter — APPROVED** [review]

**Review by siliconbrain — APPROVED** [review]
lgtm

**Pull request merged**